src/dmd/root/region.d

```diff
@@ -0,0 +1,140 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Region storage allocator implementation.
+ *
+ * Copyright:   Copyright (C) 2019-2019 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/root/region.d, root/_region.d)
+ * Documentation:  https://dlang.org/phobos/dmd_root_region.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/root/region.d
+ */
+
+module dmd.root.region;
+
+import core.stdc.string;
+import core.stdc.stdlib;
+
+import dmd.root.rmem;
+
+/*****
+ * Simple region storage allocator.
+ */
+struct Region
+{
+    void* head;         // beginning of first pool
+    void* last;         // beginning of last pool
+    void[] available;   // available to allocate
+
+    enum ChunkSize = 4096;
+    enum OverheadSize = (void*).sizeof;
+    enum MaxAllocSize = ChunkSize - OverheadSize;
+
+  nothrow:
+
+    /******
+     * Allocate nbytes. Aborts on failure.
+     * Params:
+     *  nbytes = number of bytes to allocate, can be 0, must be <= than MaxAllocSize
+     * Returns:
+     *  allocated data, null for nbytes==0
+     */
+    void* malloc(size_t nbytes)
```

Comment:
if you call it `alloc` instead of `malloc` you don't need to do `.malloc` later.

Reply:
I prefer naming it malloc, as that means allocate uninitialized data.